src/doc/src/reference/config.md

```diff
@@ -1279,6 +1280,16 @@ Can be overridden with the `--color` command-line option.
 
 Controls whether or not hyperlinks are used in the terminal.
 
+#### `term.charset`
+* Type: string
+* Default: auto-detect
+* Environment: `CARGO_TERM_CHARSET`
+
+Sets what characters can be used when rendering cargo output.  Possible values:
+
+* `utf8`: Use the full set of UTF-8 characters for drawing
+* `ascii`: Only use 7-bit ASCII characters for drawing
```

Comment:
My main concern is the future possibilities.  If we later add [emoji](https://crates.io/crates/terminal-supports-emoji) or [devicon fonts](https://devicon.dev/), how would we represent this?  A separate field, hierarchical values, or a list?

Reply:
One possible way: default to "auto" (I sympathize this word after cargo meeting today), and remove the `utf8` option. This could maximize our future compatibility.

Reply:
The downside is that doesn't allow forcing of `utf8` if  our `auto` is being overly selective.  Granted, maybe thats not too bad for now?  It would mean we'd need separate enums for config from internal / CLI to support `cargo tree --charset utf8`.

Thoughts?

Reply:
Do we have any incoming feature going to use Unicode tables? If not, I feel like stabilizing fewer things buys us more time to deal with future changes.

BTW, I was about to convince myself an “auto” option is more aligned to other `term` configs, until I saw `term.hyperlinks`.

>  It would mean we'd need separate enums for config from internal / CLI to support cargo tree --charset utf8.

Yeah. Sounds like it 😞.

Reply:
Yes, there isn't something using this yet.  Overall, this is part of a slow roll out of features to unblock user control over more complex progress bars so we have cleaner command output.

I also don't see us evaluating devicons any time soon.  Emoji is hit or miss. I don't see an immediate use of it but personally not a big emoji person.


Maybe we can at least hold off on this until I get a little further into the progress bar implementation research?  That way if emoji somehow looks useful in that time we re-evaluate?

Reply:
I just re-worked this PR to instead have a `term.unicode` field.  This would allow us to add more fields in the future like `term.emoji` without worrying about the specifics of what we might need in the future.

Thoughts?